hazelcast/src/test/java/com/hazelcast/internal/partition/membergroup/MemberGroupFactoryTest.java

```diff
@@ -66,12 +66,7 @@ public void testHostAwareMemberGroupFactoryCreateMemberGroups() {
     public void testZoneAwareMemberGroupFactoryThrowsIllegalArgumentExceptionWhenNoMetadataIsProvided() {
         MemberGroupFactory groupFactory = new ZoneAwareMemberGroupFactory();
         Collection<Member> members = createMembersWithNoMetadata();
-        Collection<MemberGroup> memberGroups = groupFactory.createMemberGroups(members);
-
-        assertEquals("Member Groups: " + String.valueOf(memberGroups), 3, memberGroups.size());
-        for (MemberGroup memberGroup : memberGroups) {
-            assertEquals("Member Group: " + String.valueOf(memberGroup), 1, memberGroup.size());
-        }
+        groupFactory.createMemberGroups(members);
```

Comment:
Minor: maybe we can use Junit5's `assertThrows()` instead of the `@Test(expected...)` to ensure the exception is thrown where we expect:
```
assertThrows(IllegalArgumentException.class, () -> groupFactory.createMemberGroups(members));
```